### ISSUE TITLE
Line Spacing and Paragraph Spacing

### ISSUE THREAD

**Issue by nasog** [hexchat/hexchat#546]
An option to change the spacing of lines and paragraphs without having to edit the source code would be nice.

**Comment by RichardHitt** [comment]
I have an alpha version of this enhancement that I'm currently using.  If you feel you can build hexchat from source and you would like to participate in a little pre-Alpha testing, do give a holler here.

**Comment by ironcladlou** [comment]
@RichardHitt 
>I have an alpha version of this enhancement that I'm currently using. If you feel you can build hexchat from source and you would like to participate in a little pre-Alpha testing, do give a holler here.

More details? I'm interested in trying to build this for Fedora 21.

**Comment by RichardHitt** [comment]
Hi, and thanks!  You can clone my latest preAlpha with this command:

     git clone -b 546PreAlpha http://github.com/RichardHitt/hexchat

This clone defaults to a value of 3 (vertical pixels) of additional 
space after each message in the text window.  Change it with "/set 
text_spacing_msg N" and after that, do "/gui apply".  Please let me know 
your experience, positive and negative.

Richard

**Comment by ironcladlou** [comment]
@RichardHitt 
>Hi, and thanks! You can clone my latest preAlpha with this command: git clone -b 546PreAlpha http://github.com/RichardHitt/hexchat This clone defaults to a value of 3 (vertical pixels) of additional space after each message in the text window. Change it with "/set text_spacing_msg N" and after that, do "/gui apply". Please let me know your experience, positive and negative. Richard

I tested the line spacing build on:

`Linux thinkpad.local 3.18.7-200.fc21.x86_64 #1 SMP Wed Feb 11 21:53:17 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux`

The setting works really well. Thank you. It's a great improvement.

Aside: It's a shame X/HexChat doesn't just render to an HTML view and support CSS theming like LimeChat (and derivatives like Textual) on OSX. The latter are extremely readable thanks to CSS theming support which allows totally independent style configurations for elements (message, timestamp, etc) as well as control over things like the column separator and scrollback marker. If anybody knows of any linux clients that support that kind of thing, I'd love to hear about them.

**Comment by TingPing** [comment]
>Aside: It's a shame X/HexChat doesn't just render to an HTML view and support CSS theming like LimeChat (and derivatives like Textual) on OSX.

They also have extremely bad performance issues using large amounts of CPU and Memory.

**Comment by walliski** [comment]
Why isn't this a thing yet :( ?

**Comment by Jayflux** [comment]
I agree, i would also like this!

**Comment by cazandobits** [comment]
@RichardHitt could you make your modifications to the last stable version ??

**Comment by TingPing** [comment]
@cazandobits Sorry but Richard passed away a few months ago.

**Comment by vadosnaprimer** [comment]
Is this waiting for a pull request or just ignored for good?

**Comment by Arnavion** [comment]
All open issues are implicitly waiting for pull requests.

**Comment by DrReD** [comment]
Any chance the changes made by RichardHitt get pulled in mainline? (http://github.com/RichardHitt/hexchat)

**Comment by cajuncooks** [comment]
I (hand-)merged the RichardHitt changes into the current hexchat/hexchat master branch on account of what we're experiencing with #2449, and unfortunately it doesn't accomplish the desired effect of not cutting off the bottom of text. I'm happy to share this branch if others would want to further maintain it for a pull, though. I ultimately had no issues with it; it works (not in the way that I expected, which is fine), except for the compilation of the Python plugin, which I think is unrelated, as I couldn't compile that plugin without pulling in his changes either.

**Comment by Apachez-** [comment]
TLDR: Until Hexchat gets support of spaces between chatlines a possible workaround is to select and use "Noto Sans Mono 9" as the font instead of "Monospace 9".

Longer read:

I experienced this issue myself after upgrading from Ubuntu 19.10 into Ubuntu 20.04. Dunno what Hexchat version it was previously but now its 2.14.3-3 and it got this issue of not enough of space between chatlines. Also the bottom line (last chat line) gets a few pixels cut off so a g looks more like a q.

So the proper fix would be if Hexchat got support of a setting where you can add additional empty pixels between the lines (including the bottom one).

A workaround that I found was that the "Monospace 9" which Hexchat in Ubuntu defaults to depends on what (currently) Gnome as window manager defines as the monospace font. Dunno if that changed between Ubuntu 19.10 and 20.04 (in 20.04 it is Ubuntu Mono 13 configurable through Gnome Tweaks app) but when changing this into "Noto Sans Mono 9" in Hexchat I got the old look back (well as it looked like in Ubuntu 19.10) including NOT getting the last line cut off at the bottom.

So I think the bug is twofolded. One is the space between chatlines but also space between the last chatline (at bottom) and the gui border. A quickfix for the Hexchat developers regarding that last chatline would be if its possible to always add an empty chatline beneath the last chatline to get the spacing needed.

Or better yet define that in pixels so basically two global settings: "/set text_spacing_msg N" and "/set
text_spacing_lastmsg N".

**Comment by foresto** [comment]
This also affects descenders on various Microsoft core fonts. For example, with Tahoma Regular 10, the bottom of the lowercase g is truncated, making it look like a q.  (Tested with bytecode hinting enabled and antialiasing disabled, as is appropriate for that font.)
